src/representation_theory/group_cohomology_resolution.lean

```diff
@@ -372,4 +430,141 @@ begin
+      (finsupp.map_domain _ (finsupp.single _ _)) = finsupp.total _ _ _ _ (finsupp.single _ _),
+    simp only [finsupp.map_domain_single, finsupp.total_single],
+  end }
+
+/-- The homotopy equivalence of complexes of `k`-modules between the standard resolution of `k` as
+a trivial `G`-representation, and the complex which is `k` at 0 and 0 everywhere else, acts as
+`∑ nᵢgᵢ ↦ ∑ nᵢ : k[G¹] → k` at 0. -/
+lemma forget₂_to_Module_homotopy_equiv_f_0_eq :
+  (forget₂_to_Module_homotopy_equiv k G).1.f 0 =
+  (forget₂ (Rep k G) _).map (π k G) :=
+begin
+  show (homotopy_equiv.hom _ ≫ (homotopy_equiv.hom _ ≫ homotopy_equiv.hom _)).f 0 = _,
+  simp only [homological_complex.comp_f],
+  convert category.id_comp _,
+  { dunfold homotopy_equiv.of_iso comp_forget_augmented_iso map_alternating_face_map_complex,
+    simp only [iso.symm_hom, eq_to_iso.inv, homological_complex.eq_to_hom_f, eq_to_hom_refl] },
+  transitivity ((finsupp.total _ _ _ (λ f, (1 : k))).comp
+    ((Module.free k).map (terminal.from _))),
+  { show _ = linear_map.comp _ (finsupp.lmap_domain _ _ _),
+    rw [@finsupp.lmap_domain_total (fin 1 → G) k k _ _ _ (⊤_ (Type u))
+      k _ _ (λ i, (1 : k)) (λ i, (1 : k)) (terminal.from ((classifying_space_universal_cover G).obj
+      (opposite.op (simplex_category.mk 0))).V) linear_map.id (λ i, rfl), linear_map.id_comp],
+    refl },
+  { congr,
+    { ext,
+      show finsupp.total _ k k _ (finsupp.single _ (1 : k)) = finsupp.single a 1 _,
+      rw [finsupp.total_single, one_smul, @unique.eq_default _
+        types.terminal_iso.to_equiv.unique a, finsupp.single_eq_same] },
+    { exact (@subsingleton.elim _ (@unique.subsingleton _ (limits.unique_to_terminal _)) _ _) }}
+end
+
+lemma d_comp_π : (group_cohomology.resolution k G).d 1 0 ≫ π k G = 0 :=
+begin
+  ext1,
+  refine linear_map.ext (λ x, _),
+  have : (forget₂_to_Module k G).d 1 0 ≫ (forget₂ (Rep k G) (Module.{u} k)).map (π k G) = 0,
+  by rw [←forget₂_to_Module_homotopy_equiv_f_0_eq,
+    ←(forget₂_to_Module_homotopy_equiv k G).1.2 1 0 rfl]; exact comp_zero,
+  exact linear_map.ext_iff.1 this _,
+end
+
+theorem forget₂_to_Module_exact_succ (n : ℕ) :
+  exact ((group_cohomology.resolution.forget₂_to_Module k G).d (n + 2) (n + 1))
+    ((group_cohomology.resolution.forget₂_to_Module k G).d (n + 1) n) :=
+(preadditive.exact_iff_homology_zero _ _).2
+  ⟨(group_cohomology.resolution.forget₂_to_Module k G).d_comp_d _ _ _,
+  ⟨(chain_complex.homology_succ_iso _ _).symm.trans ((homology_obj_iso_of_homotopy_equiv
+  (forget₂_to_Module_homotopy_equiv k G) _).trans homology_zero_zero)⟩⟩
+
+theorem exact_at_succ (n : ℕ) :
+  exact ((group_cohomology.resolution k G).d (n + 2) (n + 1))
+    ((group_cohomology.resolution k G).d (n + 1) n) :=
+(forget₂ (Rep k G) (Module.{u} k)).exact_of_exact_map
+  (forget₂_to_Module_exact_succ _ _ _)
+
+section homotopy_equiv_single₀
+
+/- Not sure where to put these at the moment; algebra/homology/homotopy doesn't import
+algebra/homology/single and vice versa. -/
```

Comment:
I would suggest showing more generally that for (co)chain complexes, the (co)homology at zero is a (co)kernel (in `algebra.homology.homology`). Then, use `homotopy_equiv.to_quasi_iso`, the isomorphism in homology given by a quasi_iso and `homology_functor_0_single₀`.

Reply:
Sorry to be dense, but just to check: for the first bit, should I be defining something like
`example : colimit_cocone (parallel_pair (C.d 1 0) 0) :=
{ X := C.homology 0,
  ι := ... }`?

Reply:
I was thinking of `def chain_complex.homology_at_zero_iso_cokernel (C : chain_complex V ℕ) [has_cokernels V] [has_kernels V] [has_images V] : homological_complex.homology C 0 ≅ cokernel (C.d 1 0) := sorry`. I am not completely sure how to prove this (using the properties of homology, or the universal property of cokernel as you suggest). It might be difficult with the current homology API. If this turns out to be too complicated (or very significantly longer than your first approach), it might be reasonable to keep your original code until the homology API is refactored.

Reply:
Ah, I see - that should be fine, I have defined it in the past. Thank you!

Reply:
Hm, I've implemented this because it should definitely be in the library but I think it's still easier to show `forget_to_Module_exact₀` using the previous way. I'm going to add a bit more API in `category_theory/preadditive/projective_resolution.lean` for showing that a complex of projectives which is homotopy equiv to `chain_complex.single₀ X` is a projective resolution of X.